jsapp/js/dataInterface.ts

```diff
@@ -1688,7 +1736,7 @@ export const dataInterface: DataInterface = {
     });
   },
 
-  loadNextPageUrl(nextPageUrl: string): JQuery.jqXHR<any> {
+  loadNextPageUrl<T>(nextPageUrl: string): JQuery.jqXHR<PaginatedResponse<T>> {
```

Comment:
```suggestion
  loadNextPageUrl(nextPageUrl: string): JQuery.jqXHR<PaginatedResponse<unknown>> {
```
> The type parameter T only appears once, so it must be bad. How to fix it? It depends what your goal is. These so-called "return-only generics" are dangerous because they're equivalent to any, but don't use the word any
> https://www.webdevtutor.net/blog/typescript-generic-type-parameter

Reply:
I've added this so that in https://github.com/kobotoolbox/kpi/pull/5453/files#diff-4aa88006757ec93f9c002b67d122faaa95cba012a3d3f30665e897a2aeb60278R106 I can get proper type as response data.

See if this commit makes it better: https://github.com/kobotoolbox/kpi/pull/5453/commits/6136967f8cd7a168f4dddcaf642443a9c1a054d8

Reply:
oops sorry, wrong link :facepalm: here's the right one https://effectivetypescript.com/2020/08/12/generics-golden-rule/